### ISSUE TITLE
[PSCore on Windows] Pasting with right-click removes an em-dash when PSReadLine is imported

### ISSUE THREAD

**Issue by alexandair** [PowerShell/PSReadLine#760]
This issue exists only in PowerShell Core console on Windows when PSReadLine in imported and **right-click** is used to paste the code that contains an em-dash. Also, CTRL+V (with an em-dash in code) works in PowerShell Core on Windows.

I had `$env:PSModulePath –split ';'` copy/pasted from somewhere and wanted to paste it to PowerShell Core console on Windows. I didn't noticed that an am-dash was used. Pasting with the right-click removed an am-dash. The same happens to smart quotes.

Before: `$env:PSModulePath –split ‘;’`
After: `$env:PSModulePath split ;`

Environment data
----------------
```powershell
PS version: 6.1.0
PSReadline version: 2.0.0-beta3
os: 10.0.17134.1 (WinBuild.160101.0800)
PS file version: 6.1.0
```
Steps to reproduce or exception report
--------------------------------------
Paste `$env:PSModulePath –split ';'` with the right-click in PowerShell Core console on Windows.

**Comment by msftrncs** [comment]
@alexandair, Its striping figure-dash, en-dash, em-dash, horizontal bar and all the curly quotes as well.   It accepts them only if its the first character on the line.  If I unload PSReadLine then I can paste them on the prompt just fine.

Sample of the characters, that if appear anywhere other than at the start of the paste, are stripped.
```
‒–—―
 “”„  ‟ 
‛’‚‘
```

I was testing this on PowerShell 5.1 and 6.1 with PSReadLine 2.0.0 loaded, testing which ones were supported by PowerShell.  Surprisingly the quote U+201F and the figure-dash are not handled as it would seem they should be.  Horizontal bar is accepted as a dash though.

**Comment by daxian-dbw** [comment]
Resolve this one as a duplicate of #1603 (forgot this issue when #1603 was opened).

Investigation shows this is likely caused by the call to `Console.KeyAvailable`. It somehow causes the next Unicode character input to be skipped. I opened https://github.com/dotnet/runtime/issues/38966 and will mark this issue duplicate/external.
